### PR TITLE
Style

### PR DESCRIPTION
**Pull request by cor3ntin**
Clean the code formatting - on top of the Qt5 branch which should be merged first

### CODE REVIEW

File: clients/replay/play.cpp [hunk 1]
```diff
@@ -163,56 +162,56 @@ namespace Aseba
...
-void dumpHelp(std::ostream &stream, const char *programName)
+void dumpHelp(std::ostream& stream, const char* programName)
 {
 	stream << "Aseba play, play recorded user messages from a file or stdin, usage:\n";
 	stream << programName << " [options] [targets]*\n";
 	stream << "Options:\n";
-	stream << "--fast          : replay messages twice the speed of real time\n";
-	stream << "--faster        : replay messages four times the speed of real time\n";
-	stream << "--fastest       : replay messages as fast as possible\n";
+	stream << "--fast		   : replay messages twice the speed of real time\n";
```
Comment (stephanemagnenat) [inline clients/replay/play.cpp]: this is a feature breakage

File: clients/studio/ConfigDialog.cpp [hunk 2]
```diff
@@ -23,36 +23,38 @@
...
-		if (me) \
-			return me->page->checkboxCache[#keyword].value; \
-		else \
-			return false; \
-	} \
-	\
-	void ConfigDialog::set##name(bool value) { \
-		if (me && !me->isVisible()) \
-			me->page->checkboxCache[#keyword].value = value; \
+	const bool ConfigDialog::get##name()                      \
```
Comment (stephanemagnenat) [inline clients/studio/ConfigDialog.cpp]: Why the spacing before the \ ?

File: clients/studio/ConfigDialog.h [hunk 3]
```diff
@@ -120,26 +123,30 @@ namespace Aseba
 		QCheckBox* newCheckbox(QString label, QString ID, bool checked = false);
 
 	protected:
-		template <class T> struct WidgetCache {
+		template<class T>
+		struct WidgetCache
+		{
 			WidgetCache() {}
-			WidgetCache(QWidget* widget, T value):widget(widget),value(value){}
+			WidgetCache(QWidget* widget, T value) :
```
Comment (stephanemagnenat) [inline clients/studio/ConfigDialog.h]: Can we configure the style tool to not change the style for constructors? The nicest depends on the context.

File: clients/studio/CustomDelegate.h [hunk 4]
```diff
@@ -27,26 +27,27 @@ namespace Aseba
...
-			const QModelIndex &index) const;
-	
-		void setEditorData(QWidget *editor, const QModelIndex &index) const;
-		void setModelData(QWidget *editor, QAbstractItemModel *model, const QModelIndex &index) const;
-	
-		void updateEditorGeometry(QWidget *editor, const QStyleOptionViewItem &option, const QModelIndex &index) const;
+		SpinBoxDelegate(int minValue, int maxValue, QObject* parent = 0);
+
+		QWidget* createEditor(QWidget* parent, const QStyleOptionViewItem& option,
+							  const QModelIndex& index) const;
```
Comment (stephanemagnenat) [inline clients/studio/CustomDelegate.h]: Why so much spacing?

File: clients/studio/DashelTarget.cpp [hunk 5]
```diff
@@ -508,73 +512,71 @@ namespace Aseba
...
+								 QApplication::tr("Protocol version mismatch"),
+								 QApplication::tr("Aseba Studio uses an older protocol (%1) than node %0 (%2), please upgrade Aseba Studio.").arg(QString::fromStdWString(nodeName.c_str())).arg(ASEBA_PROTOCOL_VERSION).arg(protocolVersion));
 		}
 		else if (protocolVersion < ASEBA_PROTOCOL_VERSION)
 		{
 			QMessageBox::warning(0,
-				QApplication::tr("Protocol version mismatch"),
-				QApplication::tr("Node %0 uses an older protocol (%2) than Aseba Studio (%1), please upgrade the node firmware.").arg(QString::fromStdWString(nodeName.c_str())).arg(ASEBA_PROTOCOL_VERSION).arg(protocolVersion)
-			);
+								 QApplication::tr("Protocol version mismatch"),
```
Comment (stephanemagnenat) [inline clients/studio/DashelTarget.cpp]: again, this spacing is wrong....

File: clients/studio/DashelTarget.cpp [hunk 6]
```diff
@@ -932,108 +934,108 @@ namespace Aseba
...
-//		}
-	}
-	
-	void DashelTarget::receivedExecutionStateChanged(Message *message)
-	{
-		ExecutionStateChanged *ess = polymorphic_downcast<ExecutionStateChanged *>(message);
-		
-		Node &node = nodes[ess->source];
+		//		if (line >= 0)
+		//		{
```
Comment (stephanemagnenat) [inline clients/studio/DashelTarget.cpp]: weird spacing, the return...

File: clients/studio/DashelTarget.cpp [hunk 7]
```diff
@@ -1076,9 +1078,8 @@ namespace Aseba
 					else if (node.steppingInNext == WAITING_LINE_CHANGE)
 					{
 						if (
-							(node.eventAddressToId.find(ess->pc) != node.eventAddressToId.end()) ||
-							(line != static_cast<int>(node.lineInNext))
-						)
+						  (node.eventAddressToId.find(ess->pc) != node.eventAddressToId.end()) ||
+						  (line != static_cast<int>(node.lineInNext)))
```
Comment (stephanemagnenat) [inline clients/studio/DashelTarget.cpp]: I think the ) should close on its own line for the sake of symetry.

File: clients/studio/EventViewer.cpp [hunk 8]
```diff
@@ -37,122 +37,145 @@
...
 	public:
 		EventDataWrapper(std::deque<double>& _x, std::deque<int16_t>& _y) :
 			_x(_x),
 			_y(_y)
-		{ }
-		virtual QwtData *   copy () const { return new EventDataWrapper(*this); }
-		virtual size_t   size () const { return _x.size(); }
-		virtual double x (size_t i) const { return _x[i]; }
-		virtual double y (size_t i) const { return (double)_y[i]; }
+		{
```
Comment (stephanemagnenat) [inline clients/studio/EventViewer.cpp]: As with Dashel, I prefer to keep these on one line...

File: clients/studio/NamedValuesVectorModel.cpp [hunk 9]
```diff
@@ -415,10 +408,10 @@ namespace Aseba
 		return true;
 	}
 
-	void MaskableNamedValuesVectorModel::toggle(const QModelIndex &index)
+	void MaskableNamedValuesVectorModel::toggle(const QModelIndex& index)
 	{
 		Q_ASSERT(namedValues);
-		Q_ASSERT(index.row() < (int)namedValues->size());
+		Q_ASSERT(index.row() < ( int )namedValues->size());
```
Comment (stephanemagnenat) [inline clients/studio/NamedValuesVectorModel.cpp]: do we need space for this simple case?

File: clients/studio/plugins/ThymioBlockly/ThymioBlockly.h [hunk 10]
```diff
@@ -39,97 +39,101 @@
 
 #include "AsebaJavascriptInterface.h"
 
-namespace Aseba { namespace ThymioBlockly
```
Comment (stephanemagnenat) [inline clients/studio/plugins/ThymioBlockly/ThymioBlockly.h]: This namespace change breaks all the code, I think it is not worth it, can we disable this style constraint?

File: common/about/AboutDialog.cpp [hunk 11]
```diff
@@ -80,24 +80,22 @@ namespace Aseba
...
-			tr("Aseba is open-source licensed under the <a href=\"https://www.gnu.org/licenses/lgpl.html\">LGPL version 3</a>.") + "</p>"
-		;
+		  "<p>" + parameters.description + "</p>" +
+		  (parameters.helpUrl.isEmpty() ? "" : "<p>" + tr("You can find more information online at <a href=\"%1\">%2</a>.").arg(parameters.helpUrl.toString()).arg(parameters.helpUrl.toString(QUrl::RemoveScheme | QUrl::RemoveUserInfo | QUrl::StripTrailingSlash).remove(0, 2)) + "</p>") +
+		  "<p>" + tr("This program is part of Aseba, a set of tools which allow beginners to program robots easily and efficiently. For more information about Aseba, visit %1.").arg("<a href=\"http://aseba.io\">aseba.io</a>") + "</p>" +
+		  "<p>" + tr("(c) 2006-2017 <a href=\"http://stephane.magnenat.net\">Stéphane Magnenat</a> and <a href=\"https://github.com/aseba-community/aseba/blob/master/authors.txt\">other contributors</a>. See tabs \"Authors\" and \"Thanks To\" for more information.") + " " +
+		  tr("Aseba is open-source licensed under the <a href=\"https://www.gnu.org/licenses/lgpl.html\">LGPL version 3</a>.") + "</p>";
```
Comment (stephanemagnenat) [inline common/about/AboutDialog.cpp]: I like the semi-colon on its own line for the sake of symmetry.

File: common/about/AboutDialog.cpp [hunk 12]
```diff
@@ -192,69 +187,68 @@ namespace Aseba
...
 		const QString dashelStreamInfo = tr("supported stream types: %1").arg(QString::fromStdString(Dashel::streamTypeRegistry.list()));
 		const QString liStart = "<li style=\"margin-top:5px; margin-bottom:5px;\">";
 		const QString libraryText =
-			welcomeText + "<ul>" +
-			liStart + libEntryText.arg("Aseba").arg(ASEBA_VERSION).arg("http://aseba.io") + "<br/>" +
-			asebaBuildInfo + "</li>" + 
-			liStart + libEntryText.arg("Dashel").arg(DASHEL_VERSION).arg("http://aseba-community.github.io/dashel/") + "<br/>" +
-			dashelStreamInfo + "</li>" +
-			liStart + libEntryText.arg("Qt").arg(qVersion()).arg("https://www.qt.io/") + "</li>" +
+		  welcomeText + "<ul>" +
```
Comment (stephanemagnenat) [inline common/about/AboutDialog.cpp]: why two spaces here?

File: common/authors.h [hunk 13]
```diff
@@ -26,57 +26,60 @@ namespace Aseba
 	using InstitutionInfos = std::vector<InstitutionInfo>;
 
 	static const AuthorInfos authorList = {
-		{
```
Comment (stephanemagnenat) [inline common/authors.h]: Do not apply styling here, it is generated code.

File: common/msg/NodesManager.h [hunk 14]
```diff
@@ -30,74 +30,74 @@ namespace Aseba
...
 			bool connected{true}; //!< whether this node is considered connected, on a "physical" level
 			UnifiedTime lastSeen; //!< when this node was last seen?
-			
+
 			bool isComplete() const;
 		};
 		//! Map from nodes id to nodes descriptions
 		typedef std::map<unsigned, Node> NodesMap;
-		NodesMap nodes; //!< all known nodes descriptions and connection status
+		NodesMap nodes;						 //!< all known nodes descriptions and connection status
```
Comment (stephanemagnenat) [inline common/msg/NodesManager.h]: buggy spacing

File: common/zeroconf/avahi-compat-libdns_sd/compat.c [hunk 15]
```diff
@@ -49,1507 +49,1626 @@
 #include "warn.h"
 #include "dns_sd.h"
 
-enum {
```
Comment (stephanemagnenat) [inline common/zeroconf/avahi-compat-libdns_sd/compat.c]: Do not restyle third-party files...

File: common/zeroconf/avahi-compat-libdns_sd/warn.h [hunk 16]
```diff
@@ -21,16 +21,29 @@
 ***/
 
 /* This routine works on Linux only, so don't rely on it */
-const char *avahi_exe_name(void);
```
Comment (stephanemagnenat) [inline common/zeroconf/avahi-compat-libdns_sd/warn.h]: Do not restyle third-party code.

File: common/zeroconf/target.cpp [hunk 17]
```diff
@@ -38,41 +38,42 @@ namespace Aseba
...
-		;
+		return lhs.name == rhs.name &&
+			   lhs.regtype == rhs.regtype &&
+			   lhs.domain == rhs.domain;
 	}
 
 	//! This target is described by a human-readable name, regtype, and domain.
 	//! It corresponds to a remote target on which that we want to resolve
-	Zeroconf::TargetInformation::TargetInformation(std::string name, std::string regtype, std::string domain):
+	Zeroconf::TargetInformation::TargetInformation(std::string name, std::string regtype, std::string domain) :
```
Comment (stephanemagnenat) [inline common/zeroconf/target.cpp]: Why spacing here?

File: common/zeroconf/target.cpp [hunk 18]
```diff
@@ -38,41 +38,42 @@ namespace Aseba
...
 		Zeroconf::TargetKey{std::move(name), std::move(regtype), std::move(domain)}
-	{}
+	{
```
Comment (stephanemagnenat) [inline common/zeroconf/target.cpp]: Having the {} on a single line is nice, can we leave it?

File: compiler/compiler.h [hunk 19]
```diff
@@ -300,19 +328,34 @@ namespace Aseba
...
-		void setTargetDescription(const TargetDescription *description);
-		const TargetDescription *getTargetDescription() const { return targetDescription;}
-		const VariablesMap *getVariablesMap() const { return &variablesMap; }
-		const SubroutineTable *getSubroutineTable() const { return &subroutineTable; }
-		void setCommonDefinitions(const CommonDefinitions *definitions);
-		bool compile(std::wistream& source, BytecodeVector& bytecode, unsigned& allocatedVariablesCount, Error &errorDescription, std::wostream* dump = nullptr);
-		void setTranslateCallback(ErrorMessages::ErrorCallback newCB) { TranslatableError::setTranslateCB(newCB); }
-		static std::wstring translate(ErrorCode error) { return TranslatableError::translateCB(error); }
+		void setTargetDescription(const TargetDescription* description);
+		const TargetDescription* getTargetDescription() const
```
Comment (stephanemagnenat) [inline compiler/compiler.h]: This decompactification hampers readability.

File: compiler/errors.cpp [hunk 20]
```diff
@@ -27,161 +27,161 @@ namespace Aseba
...
 	/** \addtogroup compiler */
 	/*@{*/
-	
+
 	static const wchar_t* error_map[ERROR_END];
 
 	ErrorMessages::ErrorMessages()
 	{
 		// compiler.cpp
-		error_map[ERROR_BROKEN_TARGET] =			L"Broken target description: not enough room for internal variables";
```
Comment (stephanemagnenat) [inline compiler/errors.cpp]: The original spacing lead to more readability.

**Comment by cor3ntin** [comment]
The PR now uses the same style than dashel.

**Pull request closed without merging**